lib/sprockets/utils.rb

```diff
@@ -90,16 +90,18 @@ def string_end_with_semicolon?(str)
     end
 
     # Internal: Accumulate asset source to buffer and append a trailing
-    # semicolon if necessary.
+    # semicolon and newline. This may result in duplicated semicolons and
+    # newlines, but we don't mind because they'll be removed by the minifier.
+    # (Previously we avoided this, but checking for existing semicolons and
+    # newlines caused significant overhead.)
     #
     # buf    - String buffer to append to
     # source - String source to append
     #
     # Returns buf String.
     def concat_javascript_sources(buf, source)
       if buf.bytesize > 0
-        buf << ";" unless string_end_with_semicolon?(buf)
```

Comment:
Yes, it looks like it can be. I've removed it in 8f9f5668bec4f98c88e3a30a0a8daf55ba6b0639